### PR TITLE
General background color UI control

### PR DESCRIPTION
**Pull request by Camwyn**
This is a "live", working, control.

The logic [here](https://github.com/moderntribe/the-events-calendar/pull/3383/files#diff-5fcdfd1965eaddaf5e3172769cf7edab8f25d45a5a0d9e667aad2d956dd404deR282) should be copied wherever we use the background color.

📷 http://p.tri.be/YLTI9F (working color option)

[TEC-3651]

[TEC-3651]: https://moderntribe.atlassian.net/browse/TEC-3651

### CODE REVIEW

File: src/Tribe/Customizer/Global_Elements.php [hunk 1]
```diff
@@ -190,22 +191,71 @@ public function register_settings( WP_Customize_Section $section, WP_Customize_M
...
+			$customizer->get_setting_name( 'background_color_choice', $section ),
+			[
+				'default' => $this->get_default( 'background_color_choice' ),
+			]
+		);
+
+		$manager->add_control(
+			$customizer->get_setting_name( 'background_color_choice', $section ),
+			[
+				'label'      => 'Background Color',
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Global_Elements.php:216]: :warning: Array double arrow not aligned correctly; expected 7 space(s) between "'label'" and double arrow, but found 6.

<sub>hash: f1fd03d1d1c26d61a999504e32b1b535</sub>

File: src/Tribe/Customizer/Global_Elements.php [hunk 2]
```diff
@@ -190,22 +191,71 @@ public function register_settings( WP_Customize_Section $section, WP_Customize_M
...
+				'section'    => $section->id,
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Global_Elements.php:217]: :warning: Array double arrow not aligned correctly; expected 5 space(s) between "'section'" and double arrow, but found 4.

<sub>hash: 4fff3509258339889905595598321c23</sub>

File: src/Tribe/Customizer/Global_Elements.php [hunk 3]
```diff
@@ -190,22 +191,71 @@ public function register_settings( WP_Customize_Section $section, WP_Customize_M
...
+				'description' => esc_html__( 'All calendar and event pages.', 'the-events-calendar'),
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Global_Elements.php:218]: * :no_entry_sign: Expected 1 spaces before closing parenthesis; 0 found
* :no_entry_sign: There must be a single space before the last parenthesis

<sub>hash: 9c9a972db38dce639de7a296086233b3</sub>

File: src/Tribe/Customizer/Global_Elements.php [hunk 4]
```diff
@@ -190,22 +191,71 @@ public function register_settings( WP_Customize_Section $section, WP_Customize_M
...
+				'type'       => 'radio',
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Global_Elements.php:219]: :warning: Array double arrow not aligned correctly; expected 8 space(s) between "'type'" and double arrow, but found 7.

<sub>hash: be5dcf847f5038f50730da451bce406e</sub>

File: src/Tribe/Customizer/Global_Elements.php [hunk 5]
```diff
@@ -190,22 +191,71 @@ public function register_settings( WP_Customize_Section $section, WP_Customize_M
...
+				'choices'    => [
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Global_Elements.php:220]: :warning: Array double arrow not aligned correctly; expected 5 space(s) between "'choices'" and double arrow, but found 4.

<sub>hash: ecea1ceb564bd5dcf0ce44d34c9dd70d</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 2 error(s)
:warning: 4 warning(s)

File: src/Tribe/Customizer/Global_Elements.php [hunk 6]
```diff
@@ -190,22 +191,70 @@ public function register_settings( WP_Customize_Section $section, WP_Customize_M
...
+				'default' => $this->get_default( 'background_color_choice' ),
+			]
+		);
+
+		$manager->add_control(
+			$customizer->get_setting_name( 'background_color_choice', $section ),
+			[
+				'label'       => 'Background Color',
+				'section'     => $section->id,
+				'description' => esc_html__( 'All calendar and event pages.', 'the-events-calendar'),
```
Comment (tr1b0t) [inline src/Tribe/Customizer/Global_Elements.php:218]: * :no_entry_sign: Expected 1 spaces before closing parenthesis; 0 found
* :no_entry_sign: There must be a single space before the last parenthesis

<sub>hash: 3cec6010bec3d8d6e2b0a52f001e42ad</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 2 error(s)

File: src/Tribe/Views/V2/Customizer.php [hunk 7]
```diff
@@ -143,16 +143,32 @@ public function filter_global_elements_css_template( $css_template, $section, $c
 			';
 		}
 
-		if ( $customizer->has_option( $section->ID, 'background_color' ) ) {
-			$css_template .= '
+		if ( $customizer->has_option( $section->ID, 'background_color_choice' )  ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:146]: :no_entry_sign: Expected exactly one space before closing parenthesis; "  " found.

<sub>hash: e95065a16ca8342dc8c74db6b98c87d7</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 8]
```diff
@@ -143,16 +143,32 @@ public function filter_global_elements_css_template( $css_template, $section, $c
...
+			if (
+				'custom' === $customizer->get_option( $section->ID, 'background_color_choice' ) &&
+				$customizer->has_option( $section->ID, 'background_color' )
+			) {
+				$css_template .= '
+					.tribe-events-view:not(.tribe-events-widget),
+					#tribe-events,
+					#tribe-events-pg-template {
+						background-color: <%= global_elements.background_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:155]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: e070dbdd871d49b50c39cd2ef3258380</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 9]
```diff
@@ -143,16 +143,32 @@ public function filter_global_elements_css_template( $css_template, $section, $c
...
+					}
+				';
+			} else {
+				$css_template .= '
 				.tribe-events-view:not(.tribe-events-widget),
 				#tribe-events,
 				#tribe-events-pg-template {
-					background-color: <%= global_elements.background_color %>;
+					background-color: <%= global_elements.background_color_choice %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:163]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: d45cb98fbcf09d452708d9bab9b2d6e3</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 1 error(s)
:warning: 2 warning(s)

File: src/Tribe/Views/V2/Customizer.php [hunk 10]
```diff
@@ -143,16 +143,32 @@ public function filter_global_elements_css_template( $css_template, $section, $c
...
 				}
 			';
+			}
```
[the remark this answers is not in the document]
Comment (Camwyn) [reply src/Tribe/Views/V2/Customizer.php:166]: Cool - wasn't 100% so I put it in.

File: src/Tribe/Customizer/Global_Elements.php [hunk 11]
```diff
@@ -190,22 +191,77 @@ public function register_settings( WP_Customize_Section $section, WP_Customize_M
...
+				[
+					'section'         => $section->id,
+					'active_callback' => function ( $control ) use ( $customizer, $section ) {
+						if (
+							'custom' == $control->manager->get_setting( $customizer->get_setting_name( 'background_color_choice', $section ) )->value()
+						) {
+							return true;
+						};
+
+						return false;
```
[the remark this answers is not in the document]
Comment (Camwyn) [reply src/Tribe/Customizer/Global_Elements.php:253]: I did and I "exploded" it for debugging as the one values saved incorrectly lead me down it's path of discovery - thanks for the reminder, I'll change it back.

**Review by lucatume — APPROVED** [review]

**Comment by paulmskim** [comment]
@Camwyn merging this so we can move ahead and dev test.

**Pull request merged**